tests/e2e/infodocs.js

```diff
@@ -0,0 +1,322 @@
+          },
+          body: {docs: docs}
+        });
+      }).then(result => {
+        // ...we expect the first two to work (because we updated their revs)...
+        assert.isTrue(result[0].ok);
+        assert.isTrue(result[1].ok);
+        // ...and the third to fail...
+        assert.isNotOk(result[2].ok);
+
+        docs[0]._rev = result[0].rev;
+        docs[1]._rev = result[1].rev;
+
+        return delayedInfoDocsOf(docs.map(d => d._id));
+      }).then(newInfoDocs => {
+        // ...which means that the first two latest_replication_date values should change...
+        assert.notEqual(newInfoDocs[0].latest_replication_date, infoDocs[0].latest_replication_date);
+        assert.notEqual(newInfoDocs[1].latest_replication_date, infoDocs[1].latest_replication_date);
+        // ..and the last should not.
+        assert.equal(newInfoDocs[2].latest_replication_date, infoDocs[2].latest_replication_date);
+
+        docs.pop(); // (remove the failing one, we're done with it)
+
+        // When we delete a doc...
+        docs[1]._deleted = true;
+
+        return utils.requestOnTestDb({
+          path: '/_bulk_docs',
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: {docs: docs}
+        });
+      }).then(result => {
+        // ...everything should work...
+        assert.isTrue(result[0].ok);
+        assert.isTrue(result[1].ok);
+
+        // ...and the second doc should not have a infodoc anymore
+        return delayedInfoDocsOf(docs[1]._id);
+      }).then(results => {
+        assert.isNull(results[0]);
+      });
+    });
+  });
+
+  describe('legacy data support', () => {
+    it('finds and migrates data from the medic doc', () => {
+      const testDoc = {
+        _id: 'yuiop',
+        data: 'data',
+        transitions: {
+          some: 'transition info'
+        }
+      };
+
+      // Prepare an existing document
+      return utils.stopSentinel()
+        .then(() => utils.db.post(testDoc))
+        .then(({id, rev}) => {
+          testDoc._id = id;
+          testDoc._rev = rev;
+
+          // Skip the preceeding write in sentinel
+          return utils.setProcessedSeqToNow();
+        })
+        .then(utils.startSentinel)
+        .then(() => {
+          // Now update the document via api and with sentinel ready to roll
+
+          testDoc.data = 'data changed';
+
+          // Should be through api
+          return utils.saveDoc(testDoc);
+        })
+        .then(() => delayedInfoDocsOf(testDoc._id))
+        .then(([infodoc]) => {
+          assert.isOk(infodoc.initial_replication_date, 'expected an initial_replication_date');
+          assert.isOk(infodoc.latest_replication_date, 'expected a latest_replication_date');
+          assert.deepEqual(infodoc.transitions, {
+            some: 'transition info'
+          });
+        });
+    });
+    it('finds and migrates data from the medic infodoc', () => {
```

Comment:
This test is actually failing on Travis: https://travis-ci.org/medic/medic/jobs/577893124 - for some reason the suite is reported as successful, but if you scroll down the tests fail because `e2e-servers` script isn't run on Travis - there services are managed by Horti :( 
Running this test locally, i get:
```
- Failed: expected {} to deeply equal { some: 'transition info' }
```